int/api/wallet/import.go

```diff
@@ -1,96 +1,72 @@
 	"github.com/massalabs/thyra/pkg/wallet"
 )
 
 const fileModeUserRW = 0o600
 
 //nolint:nolintlint,ireturn
-func NewImport(walletStorage *sync.Map) operations.MgmtWalletImportHandler {
-	return &wImport{walletStorage: walletStorage}
+func NewImport(walletStorage *sync.Map, app *fyne.App) operations.MgmtWalletImportHandler {
+	return &wImport{walletStorage: walletStorage, app: app}
 }
 
 type wImport struct {
 	walletStorage *sync.Map
+	app           *fyne.App
 }
 
 //nolint:nolintlint,ireturn,funlen
 func (c *wImport) Handle(params operations.MgmtWalletImportParams) middleware.Responder {
-	var err error
-
-	_, ok := c.walletStorage.Load(*params.Body.Nickname)
-	if ok {
-		return operations.NewMgmtWalletImportInternalServerError().WithPayload(
+	password, walletName, privateKey, err := gui.AskWalletInfo(c.app)
+	if err != nil {
+		return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
 			&models.Error{
-				Code:    errorCodeWalletAlreadyExists,
-				Message: "Error: a wallet with the same nickname already exists.",
+				Code:    errorCodeWalletCreateNew,
+				Message: err.Error(),
 			})
 	}
 
-	keyPairs := make([]wallet.KeyPair, len(params.Body.KeyPairs))
-	for index := 0; index < len(params.Body.KeyPairs); index++ {
-		keyPairs[index].PrivateKey, err = base58.CheckDecode(*params.Body.KeyPairs[index].PrivateKey)
-		if err != nil {
-			return operations.NewMgmtWalletCreateUnprocessableEntity()
-		}
-
-		keyPairs[index].PublicKey, err = base58.CheckDecode(*params.Body.KeyPairs[index].PublicKey)
-		if err != nil {
-			return operations.NewMgmtWalletCreateUnprocessableEntity()
-		}
-
-		salt, err := base58.CheckDecode(*params.Body.KeyPairs[index].Salt)
-		if err != nil {
-			return operations.NewMgmtWalletCreateUnprocessableEntity()
-		}
-
-		copy(keyPairs[index].Salt[:], salt)
-
-		nonce, err := base58.CheckDecode(*params.Body.KeyPairs[index].Nonce)
-		if err != nil {
-			return operations.NewMgmtWalletCreateUnprocessableEntity()
-		}
-
-		copy(keyPairs[index].Nonce[:], nonce)
-
-		keyPairs[index].Protected = true
+	if len(walletName) == 0 {
+		return operations.NewMgmtWalletCreateBadRequest().WithPayload(
+			&models.Error{
+				Code:    errorCodeWalletCreateNoNickname,
+				Message: "Error: nickname field is mandatory.",
+			})
 	}
 
-	newWallet := wallet.Wallet{
-		Version:  0,
-		Nickname: *params.Body.Nickname,
-		Address:  *params.Body.Address,
-		KeyPairs: keyPairs,
+	_, ok := c.walletStorage.Load(walletName)
+	if ok {
+		return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+			&models.Error{
+				Code:    errorCodeWalletAlreadyExists,
+				Message: "Error: a wallet with the same nickname already exists.",
+			})
 	}
 
-	c.walletStorage.Store(newWallet.Nickname, newWallet)
-
-	bytesOutput, err := json.Marshal(newWallet)
-	if err != nil {
-		return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+	if len(password) == 0 {
+		return operations.NewMgmtWalletCreateBadRequest().WithPayload(
```

Comment:
here the HTTP request is not involved in the password input

```suggestion
		return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
```